libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -48,6 +50,7 @@ public class EntitlementInitialization {
 
     private static final String AGENTS_PACKAGE_NAME = "co.elastic.apm.agent";
     private static final Module ENTITLEMENTS_MODULE = PolicyManager.class.getModule();
+    private static final int MINIMUM_SUPPORTED_JAVA_VERSION = 21;
```

Comment:
@rjernst do we have a better way than hardcoding this here? E.g. some other place where we already have 21 as a minimum supported version? 
Also, are we enforcing it, or we allow ES to start even with lower versions? Because if that's the case, we need to change the logic below to always include at least `org.elasticsearch.entitlement.bridge.EntitlementChecker`, or ES will fail to start on the selfTest().

Reply:
There's `JavaVersionChecker` to enforce the minimum version, @ldematte

Reply:
Thank you @mosche! TIL
Probably it's OK to have 21 hardcoded here, it's not going to change. I'd hate to add a depdendency just for this :D Wdyt?

Reply:
yes, agreed, we can't depend on this here 💯